src/daemon/rpmostree-sysroot-upgrader.c

```diff
@@ -1127,7 +1088,7 @@ overlay_packages (RpmOstreeSysrootUpgrader *self,
    * that the user wants to stop overlaying them), so we have another
    * optimization here for that case.
    */
-  if (g_strv_length (self->requested_packages) == 0)
+  if (!rpmostree_origin_is_locally_assembled (self->origin))
```

Comment:
Hmm, but the new revision can't be the base rev because creating a local initramfs changes the tree checksum, right?

Basically, I think the lowest level "overlay packages" code now needs to handle the zero length package case.  This is what I've done in my not-posted WIP for initramfs.